plugins/metrics/src/main/java/org/apache/cloudstack/metrics/MetricsServiceImpl.java

```diff
@@ -764,14 +757,13 @@ public List<ClusterMetricsResponse> listClusterMetrics(Pair<List<ClusterResponse
             if (AllowListMetricsComputation.value()) {
                 List<Ternary<Long, Long, Long>> cpuList = new ArrayList<>();
                 List<Ternary<Long, Long, Long>> memoryList = new ArrayList<>();
-                for (final Host host : hostDao.findByClusterId(clusterId)) {
-                    if (host == null || host.getType() != Host.Type.Routing) {
+                for (final HostJoinVO host : hostJoinDao.findByClusterId(clusterId, Host.Type.Routing)) {
+                    if (host == null) {
```

Comment:
I doubt a non null iterator will be returned that then refers a null-host.

Reply:
The null-check was already there, so I preserved it. But  I guess you're right, as the search criteria specifies the host type, cluster id etc. I don't think there will be a null host that has non-null type and clusterid.

Updated the commit with the removed null-check.